psql2mysql/__init__.py

```diff
@@ -203,19 +203,44 @@ def __init__(self, config, source, target, chunk_size=None):
         self.cfg = config
         self.src_uri = source if source else cfg.CONF.source
         self.target_uri = target if target else cfg.CONF.target
+        self._src_db = None
+        self._target_db = None
         try:
             self.chunk_size = positive_int(chunk_size) \
                 if chunk_size is not None else cfg.CONF.chunk_size
         except ValueError as e:
             LOG.error(str(e))
             raise
 
-    def setup(self):
-        self.src_db = DbWrapper(self.src_uri, self.chunk_size)
-        self.src_db.connect()
+    @property
+    def src_db(self):
+        if not self._src_db:
+            self._src_db = DbWrapper(self.src_uri, self.chunk_size)
+            self._src_db.connect()
+        return self._src_db
+
+    @property
+    def target_db(self):
+        if not self._target_db:
+            self._target_db = DbWrapper(self.target_uri, self.chunk_size)
+            self._target_db.connect()
+        return self._target_db
+
+    def purgeTables(self):
+        target_tables = self.target_db.getTables()
+        if not target_tables:
+            return
 
-        self.target_db = DbWrapper(self.target_uri, self.chunk_size)
-        self.target_db.connect()
+        # disable constraints on the MariaDB side for the duration of
+        # the migration
+        LOG.info("Disabling constraints on target DB for the migration")
```

Comment:
Nit: This may not be during the migration if the user only decides to clean up the tables.

Reply:
Fixed